### PR TITLE
Get Ready for Open Sourcing

### LINKED ISSUE #16: ESLint
Use same standards as other JS projects. See https://github.com/Finhaven/practices

### PR DESCRIPTION
**Pull request by expede**
# Remove Contracts
Contracts have been split out to https://github.com/Finhaven/DealToken. This brings us down to nearly a pure JS library, plus one contract for testing purposes.

# License
Added Apache 2.0, as per Boris

# Linter
I also noticed that we had some ESLint dependencies in the `package.json` that we weren't using. I enabled them and appeased the linter while debugging some tests. It's a pretty common config (Airbnb base). I can undo the style changes, but figured that they'd be nice if we're going to be open sourcing this shortly.

Closes #16 

# Readme, templates, &c
The badge won't show until this project is public. There's no point generating a token when we're about to flip it to public.

### CODE REVIEW

**Comment by expede** [comment]
@carchrae / @naumenkogs: `+4,231 −2,495` looks like a lot 😅 The actual _code_ changes are very few.

## −2,495
Mostly files moved to https://github.com/Finhaven/DealToken

## +4,231
Mostly text files, such as LICENSE, PULL_REQUEST_TEMPLATE, and so on.

File: .python-version [hunk 1]
```diff
@@ -0,0 +1 @@
+2.7.5
```
Comment (carchrae) [inline .python-version]: is this needed?  i never had this problem.

Comment (carchrae) [reply .python-version]: i remember you mentioning it before, just wonder if it is needed for anyone else - fine if it is harmless, but wonder if that is the case (eg will it restrict a person using python 2.7.4?)

File: src/lib/contract-utils.js [hunk 2]
```diff
@@ -23,13 +24,21 @@ module.exports = {
...
   },
   getDealParameters(account) {
-    const startTime = timeNow(); // web3.eth.getBlock('latest').timestamp + 1; //+ (60*60*24); //start in a day from latest block
+    // web3.eth.getBlock('latest').timestamp + 1; //+ (60*60*24); //start in a day from latest block
+    const startTime = timeNow();
     const endTime = startTime + (60 * 60 * 24 * 30); // end time, 30 days
-    const rate = web3.utils.toWei('.0001', 'ether'); // new BigNumber(1000); // rate of ether to LP Token in wei
-    const wallet = account; // the address that will hold the fund. Recommended to use a multisig one for security.
+
+    // new BigNumber(1000); // rate of ether to LP Token in wei
```
Comment (carchrae) [inline src/lib/contract-utils.js]: clean up commented out code (31 + 27)

File: src/lib/eth.js [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
-const Tx = require('ethereumjs-tx');
-const _ = require('lodash');
-const request = require('request');
 const fs = require('fs');
-const logger = console; //require('./logger');
+const request = require('request');
+const Tx = require('ethereumjs-tx');
+const Web3 = require('web3');
+
+const logger = console; // require('./logger');
```
Comment (carchrae) [inline src/lib/eth.js]: clean up commented out code

File: src/lib/eth.js [hunk 4]
```diff
@@ -279,5 +276,5 @@ try {
     .catch(e => logger.warn(`failed to connect${e}`));
 } catch (e) {
   logger.error('isConnected failed');
-  logger.warn('Ethereum node must be running for this module to work')
+  logger.warn('Ethereum node must be running for this module to work');
```
Comment (carchrae) [inline src/lib/eth.js]: may be nicer to output a hint that includes `npm run start:testrpc` or similar and also the value of the env var that specifies what eth rpc url it is using.

File: .python-version (hunk 1 above)
Comment (expede) [reply .python-version]: I'm getting this in all of my node projects. It's what I get for having Python 3 set as my default. There's a dependency that blows up unless you're on `2.7.x`.

Comment (expede) [reply .python-version]: I can gitignore it, I suppose. Just wondering it it could save someone else some grief.

**Comment by naumenkogs** [comment]
https://github.com/Finhaven/web3-server-tools/blob/050fc6e996e6c539ce966c5eb7022c1be8feb1e2/truffle/truffle.js#L7
https://github.com/Finhaven/web3-server-tools/blob/050fc6e996e6c539ce966c5eb7022c1be8feb1e2/truffle/truffle.js#L12

Should we remove these as well?

File: .python-version (hunk 1 above)
Comment (carchrae) [reply .python-version]: yeah, fine to leave it i think.  i don't know if it'll cause grief.  fwiw, i've found it easiest to keep the default OS python and pyenv/virtualenv for a peaceful pain free life.  (and rvm/npm etc)  changing default python on ubuntu breaks many things, or so i seem to recall.

**Pull request merged**